mypy/build.py

```diff
@@ -710,7 +715,7 @@ def read_with_python_encoding(path: str, pyversion: Tuple[int, int]) -> str:
             source_bytearray.decode(encoding)
         except LookupError as lookuperr:
             raise DecodeError(str(lookuperr))
-        return source_bytearray.decode(encoding)
+        return source_bytearray.decode(encoding), hashlib.md5(source_bytearray).hexdigest()
```

Comment:
Fixed in c115f139de74811d0635df50a946f3cab3a22b4c. I'm not sure why we make decision about encoding and BOM after just 2 lines rather than after reading the entire file; is it just because `source_bytearray = source_bytearray[3:]` is expensive when `source_bytearray` contains the entire file? Anyway, I assume that's the reason, so to avoid calling `f.read()` twice, I use `hashlib.md5.update()`.